### DISCUSSION TITLE
Let’s build a designer!

### DISCUSSION THREAD

**Discussion by sketchedgrey** [picoe/Eto#2005] [Ideas]
When the idea was first proposed to create a Winforms style designer 10 years ago, it didn't gain too much traction. The option to put together a small team was given (and still stands) but action was never taken. We could put together a team, but perhaps we could move something like this into a higher priority position.  

Personally, I would want this feature to be implemented because of its extreme convenience. Having a visual designer would greatly increase productivity by reducing compile times (to constantly visualise the project, especially if it was a larger application), reduce programming times, and become a better alternative to Winforms due to Eto's cross platform nature.

I'm a new contributor who hasn't submitted any pull requests at the moment, but I hope this message can start something big. I definitely DON'T have the experience to implement this by myself, but together, hey! We can build a designer!

**Comment by sketchedgrey** [comment]
Where are we going to start? Should someone fork the project or are we going to create a branch (this would be preferred for easier commit fetching).

**Comment by vesk4000** [comment]
I would love for this to happen!

**Comment by dsoronda** [comment]
Hi friends,
It's time to stop talking and start working ;)

Anyone interested please take a look : [EtoDesigner](https://github.com/dsoronda/EtoDesigner)
I'm open for pull request and ideas.

**Reply by sketchedgrey** [reply]
This is great! Sometime soon I'll write a custom toolbox/item library DLL. Thanks for kickstarting the project!

**Comment by greatawesome** [comment]
It'd be great to have a designer. There was an effort to make a cross platform WinForms designer for Mono. It is quite old and not maintained, so likely of limited use, but for reference it is here:

https://github.com/mono/mwf-designer
